docs/design/dotnet-test-for-maui.md

```diff
@@ -0,0 +1,1040 @@
+
+2. **AI Agent Enablement**: Make it easy for AI coding agents to write, run, and validate tests using only command-line tools. AI agents should be able to:
+   - Run tests with a single command
+   - Parse test results from standard output
+   - Understand which tests passed/failed
+
+3. **XHarness Replacement**: Eventually replace XHarness as the test execution tool, reducing the number of tools to maintain in the ecosystem.
+
+### 1.2 SDK Architecture
+
+This functionality will be implemented in the **Android SDK** and **iOS SDK** as first-class features. .NET MAUI will consume and leverage these SDK implementations rather than implementing its own device testing infrastructure. This ensures:
+
+- Consistent testing experience across all .NET workloads (MAUI, native iOS/Android, etc.)
+- Shared maintenance and improvements across the ecosystem
+- Direct support for native iOS and Android test projects
+- MAUI inherits all capabilities without duplication
+
+### 1.3 Phase 2 Goals (Future Work)
+
+**Code Coverage on Devices**: In a future phase, we plan to enable collection of code coverage data for platform-specific code running on iOS, Android, and Windows devices/simulators. This is **not a primary goal for Phase 1** but will be addressed later once the core testing infrastructure is established.
+
+### 1.4 Non-Goals (Phase 1)
+
+- Test Explorer integration in Visual Studio or VS Code (CLI-only focus)
+- Hot reload during test execution
+- Parallel test execution across multiple devices
+- Support every test framework in v1 (start with MSTest runner / MTP-first; broaden later)
+- Live debugging of tests in v1 (future work)
+
+---
+
+## 2. MTP Configuration
+
+### 2.1 Enabling MTP Mode
+
+`dotnet test` can run tests using **VSTest** or **Microsoft.Testing.Platform (MTP)**. For MAUI device tests, MTP mode is recommended. Enable it via `global.json`:
+
+```json
+{
+  "test": {
+    "runner": "Microsoft.Testing.Platform"
+  }
+}
+```
+
+### 2.2 MAUI Device Test Project Model
+
+A MAUI device-test project is a **MAUI app that hosts MTP**. It produces an **app package** (apk/ipa/msix) that runs tests inside the real runtime.
+
+**Key requirements:**
+- Output must be an executable test app (MTP expectation)
+- Include MTP runner + desired extensions via NuGet (TRX report, coverage)
+
+**Example project file:**
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>net10.0-android;net10.0-ios;net10.0-windows10.0.19041.0</TargetFrameworks>
+    <UseMaui>true</UseMaui>
+
+    <!-- Exe is recommended for self-hosted MTP apps (MTP itself is host-agnostic) -->
+    <OutputType>Exe</OutputType>
+    <IsTestProject>true</IsTestProject>
+
+    <!-- Device-test marker -->
+    <IsMauiDeviceTestProject>true</IsMauiDeviceTestProject>
```

Comment:
I find this redundant, ideally it relies on the SDK properties like IsTestProject.